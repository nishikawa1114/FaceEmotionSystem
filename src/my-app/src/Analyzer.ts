// import fetch from 'node-fetch';
require('dotenv').config();

export class Analyzer {

    // 画像に映る表情の分析を依頼、結果を返す
    public static analyze(imageUrl: string) {

        // Add a valid subscription key and endpoint to your environment variables.
        // const subscriptionKey = String(process.env.REACT_APP_FACE_SUBSCRIPTION_KEY);
        // const endpoint = String(process.env.REACT_APP_FACE_ENDPOINT) + '/face/v1.0/detect';
        // Optionally, replace with your own image URL (for example a .jpg or .png URL).

        // const queryParams = {    // Face API に渡すパラメータ
        //     "detectionModel": "detection_01",
        //     "returnFaceAttributes": "emotion",
        //     "returnFaceId": "true"
        // };

        // const queryString = new URLSearchParams(queryParams);

        // let fetch = require('node-fetch')
        return fetch("http://localhost:8080/face/emotion", {  // パラメータparamsをURLに入れてFace API に渡す
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({ 'url': imageUrl }) // Face API に渡したい画像のURL
        })
            .then((response) => {
                if (!response.ok) {
                    throw Error();
                }
                return response.json(); // Promiseを返す
            })
            .then((data: { [x: string]: { [x: string]: any; }; }[]) => { // JSONデータ
                // 分析結果が空の場合 
                if (data.length === 0) {
                    return Object;
                }
                // const emotion = data[0]['faceAttributes']['emotion'];
                const resuleData = data;
                return resuleData;
            })
            .catch((error: any) => { // エラーの場合
                return Object;
            });
    }
}