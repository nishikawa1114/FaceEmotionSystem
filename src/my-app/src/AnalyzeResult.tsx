import React, { useState } from 'react';
import './index.css';
import { Analyzer } from './Analyzer';
import { ImageUrl, ResultData } from './types';
import { ErrorId } from './types';
import { AnalyzeDetail } from './AnalyzeDetail';
import { AppBar, Button, CardContent, Grid, Paper, Typography } from '@material-ui/core';
import Toolbar from '@material-ui/core/Toolbar';
import Card from '@material-ui/core/Card';
import { url } from 'inspector';

interface AnalyzeProps {
    checkedimage: ImageUrl;
    onSubmit: (e: React.ChangeEvent<HTMLFormElement>) => void;
    onClick: () => void;
    setErrorId: (id: number) => void;
}

interface AnalyzeResultState {
    resultData: Array<ResultData>;
}

// 分析結果画面を表示するコンポーネント
export class AnalyzeResult extends React.Component<AnalyzeProps, AnalyzeResultState> {

    public constructor(props: AnalyzeProps) {
        super(props);
        this.state = {
            resultData: []
        }
    }

    // URLから日付を返す
    // 期待するURL
    //  ~~~/images/(ユーザー名)/(年)/(月)/(日)/(画像ファイル名)~~~ の形
    // ex) https://sample/images/nishikawa/2020/02/16/sample.jpg/~~~
    getDate = (url: string) => {
        let strDate = String(url.match(/\d{4}\/\d{2}\/\d{2}/));
        if (strDate === "null") {
            strDate = "-";
        }
        return strDate;
    }

    // URLからユーザー名を返す
    // 期待するURL
    // ~~~/images//(ユーザー名)/(年)/(月)/(日)/(画像ファイル名)~~~ の形
    // ex) https://sample/images/nishikawa/2020/02/16/sample.jpg/~~~
    getName = (url: string) => {
        const regex = /(images){1}\/.*\d{4}\/\d{2}\/\d{2}/;
        let tmp = String(url.match(regex));
        if (tmp === "null") { // images/~~ の後にユーザー名が来ることを想定しているため"images/"を確認
            return "-";
        }
        const temp: string = String(url.split('images/').pop());
        const name: string = String(temp.split('/').shift());
        return name;
    }

    // urlの画像を分析する
    private analyzeImage = async (imageUrl: string) => {
        const emotion = await Analyzer.analyze(imageUrl);
        return emotion;
    }

    public async componentDidMount() {
        this.analyzeImage(this.props.checkedimage.url).then((response: any) => {
            // 画像の分析ができない場合、APIにアクセスできない場合
            if (!Object.keys(response).length) {
                this.props.setErrorId(ErrorId.ERROR_ANALYZE_RESULT_EMPTY);
                return;
            }

            this.setState({
                resultData: response,
            })
        })
    }

    public render() {
        const MAX_ANALYSIS_DISPLAY_LENGTH = 10;

        return (
            <div className="back">

                <AppBar className="title_bar">
                    <Toolbar>
                        <Typography variant="h6" color="inherit">
                            Face Emotion System
                        </Typography>
                    </Toolbar>
                    <Toolbar>
                        <Typography variant="h6" color="inherit">
                            分析結果
                        </Typography>
                    </Toolbar>
                </AppBar>

                <Toolbar />
                <Toolbar />

                {/* 画像のユーザー名と日付を表示 */}
                <Grid className="image_info">
                    ユーザー：{this.getName(this.props.checkedimage.url)} | 日付: {this.getDate(this.props.checkedimage.url)}
                </Grid>

                {/* 分析結果を表示 */}
                {this.state.resultData.length > 0 &&
                    <div>
                        {

                            Array(this.state.resultData.slice(0, MAX_ANALYSIS_DISPLAY_LENGTH).length).fill(this.state.resultData.slice(0,10)).map((value, i: number) => {
                                return (
                                    <div key={i}>
                                        <Grid container spacing={1}>
                                            <Grid item className="analyze_grid_item">
                                                <Card>
                                                    <AnalyzeDetail
                                                        img={this.props.checkedimage}
                                                        resultData={value[i]}
                                                        id={i}
                                                    />
                                                </Card>
                                            </Grid>
                                        </Grid>
                                        <CardContent>

                                        </CardContent>

                                    </div>
                                )
                            })
                        }

                    </div>
                }
                {/* ホームへ戻るボタンの表示 */}
                <form onSubmit={this.props.onSubmit} className="home_button">
                    <Button type="submit" variant="contained" color="primary">ホーム画面へ戻る</Button>
                </form>
            </div>
        )
    }

}