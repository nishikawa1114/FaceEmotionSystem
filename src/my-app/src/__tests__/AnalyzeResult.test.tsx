import { create } from "react-test-renderer";
import React from "react";
import { AnalyzeResult } from './../AnalyzeResult';
import { Analyzer } from "./../Analyzer";
import { ImageUrl } from "../types";

// 複数人写る画像の分析結果が表示できているかテスト
describe("should display analyze results.", () => {
    let spy: jest.SpyInstance<Promise<any>, [str: string]>;


    it("display analyze result 4 pserson", (done) => {

        spy = jest.spyOn(Analyzer, 'analyze').mockImplementationOnce(
            (s: string): Promise<any> => Promise.resolve(
                {
                    "total": 4,
                    "mean": {
                        "faceAttributes": {
                            "emotion": {
                                "anger": 0,
                                "contempt": 0,
                                "disgust": 0,
                                "fear": 0,
                                "happiness": 1,
                                "neutral": 0,
                                "sadness": 0,
                                "surprise": 0
                            }
                        }
                    },
                    "person":
                        [
                            {
                                "faceID": "b4e6ce49-a1e6-492d-b9e4-f218a1720022",
                                "faceRectangle": {
                                    "top": 120,
                                    "left": 80,
                                    "width": 46,
                                    "height": 46
                                },
                                "faceAttributes": {
                                    "emotion": {
                                        "anger": 0,
                                        "contempt": 0,
                                        "disgust": 0,
                                        "fear": 0,
                                        "happiness": 1,
                                        "neutral": 0,
                                        "sadness": 0,
                                        "surprise": 0
                                    }
                                }
                            },
                            {
                                "faceID": "b43b1ad9-dab5-4b73-bf4c-285ddb1058a9",
                                "faceRectangle": {
                                    "top": 161,
                                    "left": 214,
                                    "width": 45,
                                    "height": 45
                                },
                                "faceAttributes": {
                                    "emotion": {
                                        "anger": 0.042,
                                        "contempt": 0,
                                        "disgust": 0,
                                        "fear": 0,
                                        "happiness": 0.958,
                                        "neutral": 0,
                                        "sadness": 0,
                                        "surprise": 0
                                    }
                                }
                            },
                            {
                                "faceID": "71c518d3-c68b-4664-a251-cbf9158e13b8",
                                "faceRectangle": {
                                    "top": 147,
                                    "left": 498,
                                    "width": 43,
                                    "height": 43
                                },
                                "faceAttributes": {
                                    "emotion": {
                                        "anger": 0,
                                        "contempt": 0,
                                        "disgust": 0,
                                        "fear": 0,
                                        "happiness": 1,
                                        "neutral": 0,
                                        "sadness": 0,
                                        "surprise": 0
                                    }
                                }
                            },
                            {
                                "faceID": "c8adfd69-6217-4c86-b2ae-34eb5446ac3a",
                                "faceRectangle": {
                                    "top": 112,
                                    "left": 366,
                                    "width": 41,
                                    "height": 41
                                },
                                "faceAttributes": {
                                    "emotion": {
                                        "anger": 0,
                                        "contempt": 0,
                                        "disgust": 0,
                                        "fear": 0,
                                        "happiness": 1,
                                        "neutral": 0,
                                        "sadness": 0,
                                        "surprise": 0
                                    }
                                }
                            }
                        ]
                }

            )
        );

        const img = { id: 1, url: "https://nishikawa.blob.core.windows.net/images/sugimoto/2020/11/01/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D" };
        const tree = create(
            <AnalyzeResult
                checkedimage={img}
                onClick={() => { }}
                onSubmit={() => { }}
                setErrorId={() => { }}
            />
        );
        setTimeout(() => {
            try {
                done();
                expect(tree.toJSON()).toMatchSnapshot();
            } catch (error) {
                return
            }
        }, 5000);
    })


    it("display analyze result 11 person", (done) => {


        spy = jest.spyOn(Analyzer, 'analyze').mockImplementationOnce(
            (s: string): Promise<any> => Promise.resolve(
                {
                    "total": 4,
                    "mean": {
                        "faceAttributes": {
                            "emotion": {
                                "anger": 0,
                                "contempt": 0,
                                "disgust": 0,
                                "fear": 0,
                                "happiness": 1,
                                "neutral": 0,
                                "sadness": 0,
                                "surprise": 0
                            }
                        }
                    },
                    "person":
                    [
                        {
                            "faceID": "b4e6ce49-a1e6-492d-b9e4-f218a1720022",
                            "faceRectangle": {
                                "top": 120,
                                "left": 80,
                                "width": 46,
                                "height": 46
                            },
                            "faceAttributes": {
                                "emotion": {
                                    "anger": 0,
                                    "contempt": 0,
                                    "disgust": 0,
                                    "fear": 0,
                                    "happiness": 1,
                                    "neutral": 0,
                                    "sadness": 0,
                                    "surprise": 0
                                }
                            }
                        },
                        {
                            "faceID": "b43b1ad9-dab5-4b73-bf4c-285ddb1058a9",
                            "faceRectangle": {
                                "top": 161,
                                "left": 214,
                                "width": 45,
                                "height": 45
                            },
                            "faceAttributes": {
                                "emotion": {
                                    "anger": 0.042,
                                    "contempt": 0,
                                    "disgust": 0,
                                    "fear": 0,
                                    "happiness": 0.958,
                                    "neutral": 0,
                                    "sadness": 0,
                                    "surprise": 0
                                }
                            }
                        },
                        {
                            "faceID": "71c518d3-c68b-4664-a251-cbf9158e13b8",
                            "faceRectangle": {
                                "top": 147,
                                "left": 498,
                                "width": 43,
                                "height": 43
                            },
                            "faceAttributes": {
                                "emotion": {
                                    "anger": 0,
                                    "contempt": 0,
                                    "disgust": 0,
                                    "fear": 0,
                                    "happiness": 1,
                                    "neutral": 0,
                                    "sadness": 0,
                                    "surprise": 0
                                }
                            }
                        },
                        {
                            "faceID": "c8adfd69-6217-4c86-b2ae-34eb5446ac3a",
                            "faceRectangle": {
                                "top": 112,
                                "left": 366,
                                "width": 41,
                                "height": 41
                            },
                            "faceAttributes": {
                                "emotion": {
                                    "anger": 0,
                                    "contempt": 0,
                                    "disgust": 0,
                                    "fear": 0,
                                    "happiness": 1,
                                    "neutral": 0,
                                    "sadness": 0,
                                    "surprise": 0
                                }
                            }
                        },
                        {
                            "faceID": "b4e6ce49-a1e6-492d-b9e4-f218a1720022",
                            "faceRectangle": {
                                "top": 120,
                                "left": 80,
                                "width": 46,
                                "height": 46
                            },
                            "faceAttributes": {
                                "emotion": {
                                    "anger": 0,
                                    "contempt": 0,
                                    "disgust": 0,
                                    "fear": 0,
                                    "happiness": 1,
                                    "neutral": 0,
                                    "sadness": 0,
                                    "surprise": 0
                                }
                            }
                        },
                        {
                            "faceID": "b4e6ce49-a1e6-492d-b9e4-f218a1720022",
                            "faceRectangle": {
                                "top": 120,
                                "left": 80,
                                "width": 46,
                                "height": 46
                            },
                            "faceAttributes": {
                                "emotion": {
                                    "anger": 0,
                                    "contempt": 0,
                                    "disgust": 0,
                                    "fear": 0,
                                    "happiness": 1,
                                    "neutral": 0,
                                    "sadness": 0,
                                    "surprise": 0
                                }
                            }
                        },
                        {
                            "faceID": "b4e6ce49-a1e6-492d-b9e4-f218a1720022",
                            "faceRectangle": {
                                "top": 120,
                                "left": 80,
                                "width": 46,
                                "height": 46
                            },
                            "faceAttributes": {
                                "emotion": {
                                    "anger": 0,
                                    "contempt": 0,
                                    "disgust": 0,
                                    "fear": 0,
                                    "happiness": 1,
                                    "neutral": 0,
                                    "sadness": 0,
                                    "surprise": 0
                                }
                            }
                        },
                        {
                            "faceID": "b4e6ce49-a1e6-492d-b9e4-f218a1720022",
                            "faceRectangle": {
                                "top": 120,
                                "left": 80,
                                "width": 46,
                                "height": 46
                            },
                            "faceAttributes": {
                                "emotion": {
                                    "anger": 0,
                                    "contempt": 0,
                                    "disgust": 0,
                                    "fear": 0,
                                    "happiness": 1,
                                    "neutral": 0,
                                    "sadness": 0,
                                    "surprise": 0
                                }
                            }
                        },
                        {
                            "faceID": "b4e6ce49-a1e6-492d-b9e4-f218a1720022",
                            "faceRectangle": {
                                "top": 120,
                                "left": 80,
                                "width": 46,
                                "height": 46
                            },
                            "faceAttributes": {
                                "emotion": {
                                    "anger": 0,
                                    "contempt": 0,
                                    "disgust": 0,
                                    "fear": 0,
                                    "happiness": 1,
                                    "neutral": 0,
                                    "sadness": 0,
                                    "surprise": 0
                                }
                            }
                        },
                        {
                            "faceID": "b4e6ce49-a1e6-492d-b9e4-f218a1720022",
                            "faceRectangle": {
                                "top": 120,
                                "left": 80,
                                "width": 46,
                                "height": 46
                            },
                            "faceAttributes": {
                                "emotion": {
                                    "anger": 0,
                                    "contempt": 0,
                                    "disgust": 0,
                                    "fear": 0,
                                    "happiness": 1,
                                    "neutral": 0,
                                    "sadness": 0,
                                    "surprise": 0
                                }
                            }
                        },
                        {
                            "faceID": "b4e6ce49-a1e6-492d-b9e4-f218a1720022",
                            "faceRectangle": {
                                "top": 120,
                                "left": 80,
                                "width": 46,
                                "height": 46
                            },
                            "faceAttributes": {
                                "emotion": {
                                    "anger": 0,
                                    "contempt": 0,
                                    "disgust": 0,
                                    "fear": 0,
                                    "happiness": 1,
                                    "neutral": 0,
                                    "sadness": 0,
                                    "surprise": 0
                                }
                            }
                        }
                    ]
                }
            )
        );


        const img = { id: 1, url: "https://nishikawa.blob.core.windows.net/images/sugimoto/2020/11/01/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D" };
        const tree = create(
            <AnalyzeResult
                checkedimage={img}
                onClick={() => { }}
                onSubmit={() => { }}
                setErrorId={() => { }}
            />
        );
        setTimeout(() => {
            try {
                done();
                expect(tree.toJSON()).toMatchSnapshot();
            } catch (error) {
                return
            }
        }, 5000);
    })
})

const img: ImageUrl = { id: 1, url: "https://nishikawa.blob.core.windows.net/images/steve/2020/10/15/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D" };
const props = {
    checkedimage: img,
    onSubmit: () => { },
    onClick: () => { },
    setErrorId: () => { }
}
const analyzeResult = new AnalyzeResult(props);

describe("getName() test", () => {


    it("正常動作", () => {
        const url = "https://nishikawa.blob.core.windows.net/images/nishikawa/2020/09/01/01.jpg";

        const name = analyzeResult.getName(url);
        console.log(name);
        expect(name).toEqual("nishikawa");
    })

    it("images/images/ と続く場合", () => {
        const url = "https://nishikawa.blob.core.windows.net/images/images/nishikawa/2020/09/01/01.jpg";

        const name = analyzeResult.getName(url);
        console.log(name);
        expect(name).toEqual("nishikawa");
    })

    it("images がない場合", () => {
        const url = "https://nishikawa.blob.core.windows.net/nishikawa/2020/09/01/01.jpg";

        const name = analyzeResult.getName(url);
        console.log(name);
        expect(name).toEqual("-");
    })
})


describe("getDate() test", () => {
    it("正常動作", () => {
        const url = "https://nishikawa.blob.core.windows.net/images/images/nishikawa/2020/09/01/01.jpg";

        const name = analyzeResult.getDate(url);
        console.log(name);
        expect(name).toEqual("2020/09/01");
    })

    it("日付がない場合", () => {
        const url = "https://nishikawa.blob.core.windows.net/images/nishikawa/0.jpg";

        const name = analyzeResult.getDate(url);
        console.log(name);
        expect(name).toEqual("-");
    })

    it("日付が2箇所にある場合", () => {
        const url = "https://nishikawa.blob.core.windows.net/images/images/nishikawa/2020/09/01/01.jpg/2020/01/01/";

        const name = analyzeResult.getDate(url);
        console.log(name);
        expect(name).toEqual("2020/09/01");
    })
})