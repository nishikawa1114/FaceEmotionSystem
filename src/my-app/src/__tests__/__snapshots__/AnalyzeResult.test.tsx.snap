// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display analyze result. display analyze result 1`] = `
<div>
  <header>
    <h1>
      Face Emotion System
    </h1>
  </header>
  <h2>
    分析結果
  </h2>
  <div
    className="result"
  >
    <div
      className="hidden_box"
    >
      <div
        className="image_info"
      >
        <img
          alt="dummy_url/images/steve/2020/10/15/01.jpg"
          className="image"
          onClick={[Function]}
          src="dummy_url/images/steve/2020/10/15/01.jpg"
        />
        <form>
          <input
            checked={false}
            className="image_checkbox"
            type="checkbox"
            value="dummy_url/images/steve/2020/10/15/01.jpg"
          />
        </form>
        steve
        <br />
        2020/10/15
      </div>
    </div>
    <div
      className="graph"
    >
      <div>
        <div>
          <div
            className="recharts-wrapper"
            style={
              Object {
                "cursor": "default",
                "height": 300,
                "position": "relative",
                "width": 400,
              }
            }
          >
            <svg
              className="recharts-surface"
              height={300}
              version="1.1"
              viewBox="0 0 400 300"
              width={400}
            >
              <defs>
                <clipPath
                  id="recharts2-clip"
                >
                  <rect
                    height={290}
                    width={390}
                    x={5}
                    y={5}
                  />
                </clipPath>
              </defs>
              <g
                className="recharts-polar-grid"
              >
                <g
                  className="recharts-polar-grid-angle"
                >
                  <line
                    stroke="#ccc"
                    x1={200}
                    x2={200}
                    y1={150}
                    y2={50}
                  />
                  <line
                    stroke="#ccc"
                    x1={200}
                    x2={270.71067811865476}
                    y1={150}
                    y2={79.28932188134526}
                  />
                  <line
                    stroke="#ccc"
                    x1={200}
                    x2={300}
                    y1={150}
                    y2={150}
                  />
                  <line
                    stroke="#ccc"
                    x1={200}
                    x2={270.71067811865476}
                    y1={150}
                    y2={220.71067811865476}
                  />
                  <line
                    stroke="#ccc"
                    x1={200}
                    x2={200}
                    y1={150}
                    y2={250}
                  />
                  <line
                    stroke="#ccc"
                    x1={200}
                    x2={129.28932188134524}
                    y1={150}
                    y2={220.71067811865476}
                  />
                  <line
                    stroke="#ccc"
                    x1={200}
                    x2={100}
                    y1={150}
                    y2={150}
                  />
                  <line
                    stroke="#ccc"
                    x1={200}
                    x2={129.28932188134524}
                    y1={150}
                    y2={79.28932188134526}
                  />
                </g>
                <g
                  className="recharts-polar-grid-concentric"
                >
                  <path
                    className="recharts-polar-grid-concentric-polygon"
                    d="M 200,150L 200,150L 200,150L 200,150L 200,150L 200,150L 200,150L 200,150Z"
                    fill="none"
                    stroke="#ccc"
                  />
                  <path
                    className="recharts-polar-grid-concentric-polygon"
                    d="M 200,125L 217.6776695296637,132.3223304703363L 225,150L 217.6776695296637,167.6776695296637L 200,175L 182.3223304703363,167.6776695296637L 175,150L 182.3223304703363,132.3223304703363Z"
                    fill="none"
                    stroke="#ccc"
                  />
                  <path
                    className="recharts-polar-grid-concentric-polygon"
                    d="M 200,100L 235.35533905932738,114.64466094067262L 250,150L 235.35533905932738,185.35533905932738L 200,200L 164.64466094067262,185.35533905932738L 150,150L 164.64466094067262,114.64466094067262Z"
                    fill="none"
                    stroke="#ccc"
                  />
                  <path
                    className="recharts-polar-grid-concentric-polygon"
                    d="M 200,75L 253.03300858899107,96.96699141100893L 275,150L 253.03300858899107,203.03300858899107L 200,225L 146.96699141100893,203.03300858899107L 125,150L 146.96699141100893,96.96699141100893Z"
                    fill="none"
                    stroke="#ccc"
                  />
                  <path
                    className="recharts-polar-grid-concentric-polygon"
                    d="M 200,50L 270.71067811865476,79.28932188134526L 300,150L 270.71067811865476,220.71067811865476L 200,250L 129.28932188134524,220.71067811865476L 100,150L 129.28932188134524,79.28932188134526Z"
                    fill="none"
                    stroke="#ccc"
                  />
                </g>
              </g>
              <g
                className="recharts-layer recharts-polar-angle-axis"
              >
                <polygon
                  className="recharts-polygon recharts-polar-angle-axis-line"
                  fill="none"
                  points="200,50 270.71067811865476,79.28932188134526 300,150 270.71067811865476,220.71067811865476 200,250 129.28932188134524,220.71067811865476 100,150 129.28932188134524,79.28932188134526"
                  radius={100}
                />
                <g
                  className="recharts-layer recharts-polar-angle-axis-ticks"
                >
                  <g
                    className="recharts-layer recharts-polar-angle-axis-tick"
                  >
                    <line
                      className="recharts-polar-angle-axis-tick-line"
                      fill="none"
                      radius={100}
                      x1={200}
                      x2={200}
                      y1={50}
                      y2={42}
                    />
                    <text
                      className="recharts-text recharts-polar-angle-axis-tick-value"
                      radius={100}
                      stroke="none"
                      textAnchor="middle"
                      x={200}
                      y={42}
                    >
                      <tspan
                        dy="0em"
                        x={200}
                      >
                        anger
                      </tspan>
                    </text>
                  </g>
                  <g
                    className="recharts-layer recharts-polar-angle-axis-tick"
                  >
                    <line
                      className="recharts-polar-angle-axis-tick-line"
                      fill="none"
                      radius={100}
                      x1={270.71067811865476}
                      x2={276.3675323681471}
                      y1={79.28932188134526}
                      y2={73.63246763185288}
                    />
                    <text
                      className="recharts-text recharts-polar-angle-axis-tick-value"
                      radius={100}
                      stroke="none"
                      textAnchor="start"
                      x={276.3675323681471}
                      y={73.63246763185288}
                    >
                      <tspan
                        dy="0em"
                        x={276.3675323681471}
                      >
                        contempt
                      </tspan>
                    </text>
                  </g>
                  <g
                    className="recharts-layer recharts-polar-angle-axis-tick"
                  >
                    <line
                      className="recharts-polar-angle-axis-tick-line"
                      fill="none"
                      radius={100}
                      x1={300}
                      x2={308}
                      y1={150}
                      y2={150}
                    />
                    <text
                      className="recharts-text recharts-polar-angle-axis-tick-value"
                      radius={100}
                      stroke="none"
                      textAnchor="start"
                      x={308}
                      y={150}
                    >
                      <tspan
                        dy="0em"
                        x={308}
                      >
                        disgust
                      </tspan>
                    </text>
                  </g>
                  <g
                    className="recharts-layer recharts-polar-angle-axis-tick"
                  >
                    <line
                      className="recharts-polar-angle-axis-tick-line"
                      fill="none"
                      radius={100}
                      x1={270.71067811865476}
                      x2={276.3675323681471}
                      y1={220.71067811865476}
                      y2={226.36753236814712}
                    />
                    <text
                      className="recharts-text recharts-polar-angle-axis-tick-value"
                      radius={100}
                      stroke="none"
                      textAnchor="start"
                      x={276.3675323681471}
                      y={226.36753236814712}
                    >
                      <tspan
                        dy="0em"
                        x={276.3675323681471}
                      >
                        fear
                      </tspan>
                    </text>
                  </g>
                  <g
                    className="recharts-layer recharts-polar-angle-axis-tick"
                  >
                    <line
                      className="recharts-polar-angle-axis-tick-line"
                      fill="none"
                      radius={100}
                      x1={200}
                      x2={200}
                      y1={250}
                      y2={258}
                    />
                    <text
                      className="recharts-text recharts-polar-angle-axis-tick-value"
                      radius={100}
                      stroke="none"
                      textAnchor="middle"
                      x={200}
                      y={258}
                    >
                      <tspan
                        dy="0em"
                        x={200}
                      >
                        happiness
                      </tspan>
                    </text>
                  </g>
                  <g
                    className="recharts-layer recharts-polar-angle-axis-tick"
                  >
                    <line
                      className="recharts-polar-angle-axis-tick-line"
                      fill="none"
                      radius={100}
                      x1={129.28932188134524}
                      x2={123.63246763185288}
                      y1={220.71067811865476}
                      y2={226.36753236814712}
                    />
                    <text
                      className="recharts-text recharts-polar-angle-axis-tick-value"
                      radius={100}
                      stroke="none"
                      textAnchor="end"
                      x={123.63246763185288}
                      y={226.36753236814712}
                    >
                      <tspan
                        dy="0em"
                        x={123.63246763185288}
                      >
                        neutral
                      </tspan>
                    </text>
                  </g>
                  <g
                    className="recharts-layer recharts-polar-angle-axis-tick"
                  >
                    <line
                      className="recharts-polar-angle-axis-tick-line"
                      fill="none"
                      radius={100}
                      x1={100}
                      x2={92}
                      y1={150}
                      y2={150}
                    />
                    <text
                      className="recharts-text recharts-polar-angle-axis-tick-value"
                      radius={100}
                      stroke="none"
                      textAnchor="end"
                      x={92}
                      y={150}
                    >
                      <tspan
                        dy="0em"
                        x={92}
                      >
                        sadness
                      </tspan>
                    </text>
                  </g>
                  <g
                    className="recharts-layer recharts-polar-angle-axis-tick"
                  >
                    <line
                      className="recharts-polar-angle-axis-tick-line"
                      fill="none"
                      radius={100}
                      x1={129.28932188134524}
                      x2={123.63246763185285}
                      y1={79.28932188134526}
                      y2={73.63246763185288}
                    />
                    <text
                      className="recharts-text recharts-polar-angle-axis-tick-value"
                      radius={100}
                      stroke="none"
                      textAnchor="end"
                      x={123.63246763185285}
                      y={73.63246763185288}
                    >
                      <tspan
                        dy="0em"
                        x={123.63246763185285}
                      >
                        surprise
                      </tspan>
                    </text>
                  </g>
                </g>
              </g>
              <g
                className="recharts-layer recharts-polar-radius-axis"
              >
                <line
                  className="recharts-polar-radius-axis-line"
                  fill="none"
                  radius={100}
                  stroke="#ccc"
                  x1={200}
                  x2={200}
                  y1={150}
                  y2={50}
                />
                <g
                  className="recharts-layer recharts-polar-radius-axis-ticks"
                >
                  <g
                    className="recharts-layer recharts-polar-radius-axis-tick"
                  >
                    <text
                      className="recharts-text recharts-polar-radius-axis-tick-value"
                      fill="#ccc"
                      radius={100}
                      stroke="none"
                      textAnchor="start"
                      transform="rotate(0, 200, 150)"
                      x={200}
                      y={150}
                    >
                      <tspan
                        dy="0em"
                        x={200}
                      >
                        0
                      </tspan>
                    </text>
                  </g>
                  <g
                    className="recharts-layer recharts-polar-radius-axis-tick"
                  >
                    <text
                      className="recharts-text recharts-polar-radius-axis-tick-value"
                      fill="#ccc"
                      radius={100}
                      stroke="none"
                      textAnchor="start"
                      transform="rotate(0, 200, 125)"
                      x={200}
                      y={125}
                    >
                      <tspan
                        dy="0em"
                        x={200}
                      >
                        0.25
                      </tspan>
                    </text>
                  </g>
                  <g
                    className="recharts-layer recharts-polar-radius-axis-tick"
                  >
                    <text
                      className="recharts-text recharts-polar-radius-axis-tick-value"
                      fill="#ccc"
                      radius={100}
                      stroke="none"
                      textAnchor="start"
                      transform="rotate(0, 200, 100)"
                      x={200}
                      y={100}
                    >
                      <tspan
                        dy="0em"
                        x={200}
                      >
                        0.5
                      </tspan>
                    </text>
                  </g>
                  <g
                    className="recharts-layer recharts-polar-radius-axis-tick"
                  >
                    <text
                      className="recharts-text recharts-polar-radius-axis-tick-value"
                      fill="#ccc"
                      radius={100}
                      stroke="none"
                      textAnchor="start"
                      transform="rotate(0, 200, 75)"
                      x={200}
                      y={75}
                    >
                      <tspan
                        dy="0em"
                        x={200}
                      >
                        0.75
                      </tspan>
                    </text>
                  </g>
                  <g
                    className="recharts-layer recharts-polar-radius-axis-tick"
                  >
                    <text
                      className="recharts-text recharts-polar-radius-axis-tick-value"
                      fill="#ccc"
                      radius={100}
                      stroke="none"
                      textAnchor="start"
                      transform="rotate(0, 200, 50)"
                      x={200}
                      y={50}
                    >
                      <tspan
                        dy="0em"
                        x={200}
                      >
                        1
                      </tspan>
                    </text>
                  </g>
                </g>
              </g>
              <g
                className="recharts-layer recharts-radar"
              >
                <g
                  className="recharts-layer recharts-radar-polygon"
                >
                  <polygon
                    className="recharts-polygon"
                    fill="#8884d8"
                    fillOpacity={0.8}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    points="200,140 214.14213562373095,135.85786437626905 230,150 228.2842712474619,178.2842712474619 200,200 157.57359312880715,192.42640687119285 130,150 200,150"
                    stroke="#8884d8"
                  />
                </g>
              </g>
            </svg>
            <div
              className="recharts-legend-wrapper"
              style={
                Object {
                  "bottom": 5,
                  "height": "auto",
                  "left": 5,
                  "position": "absolute",
                  "width": 390,
                }
              }
            >
              <ul
                className="recharts-default-legend"
                style={
                  Object {
                    "margin": 0,
                    "padding": 0,
                    "textAlign": "center",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div>
        <table>
          <tr>
            <th>
              感情
            </th>
            <th>
              値
            </th>
          </tr>
          <tr>
            <td
              className="emotion"
            >
              anger
            </td>
            <td
              className="emotion_value"
            >
              0.1
            </td>
          </tr>
          <tr>
            <td
              className="emotion"
            >
              contempt
            </td>
            <td
              className="emotion_value"
            >
              0.2
            </td>
          </tr>
          <tr>
            <td
              className="emotion"
            >
              disgust
            </td>
            <td
              className="emotion_value"
            >
              0.3
            </td>
          </tr>
          <tr>
            <td
              className="emotion"
            >
              fear
            </td>
            <td
              className="emotion_value"
            >
              0.4
            </td>
          </tr>
          <tr>
            <td
              className="emotion"
            >
              happiness
            </td>
            <td
              className="emotion_value"
            >
              0.5
            </td>
          </tr>
          <tr>
            <td
              className="emotion"
            >
              neutral
            </td>
            <td
              className="emotion_value"
            >
              0.6
            </td>
          </tr>
          <tr>
            <td
              className="emotion"
            >
              sadness
            </td>
            <td
              className="emotion_value"
            >
              0.7
            </td>
          </tr>
          <tr>
            <td
              className="emotion"
            >
              surprise
            </td>
            <td
              className="emotion_value"
            >
              0
            </td>
          </tr>
        </table>
      </div>
    </div>
  </div>
  <form
    className="home_button"
    onSubmit={[Function]}
  >
    <button
      type="submit"
    >
      ホーム画面へ戻る
    </button>
  </form>
</div>
`;
